tutorials/oauth/python/README.md

```diff
@@ -1,148 +1,123 @@
+Install the sample dependencies:
 
-For more information on the Azure AD OAuth 2.0 for Event Hubs see [Authorize access with Azure Active Directory](https://docs.microsoft.com/en-us/azure/event-hubs/authorize-access-azure-active-directory#overview)
+```shell
+python -m pip install -r requirements.txt
+```
 
-### Retrieve Azure Active Directory (AAD) Token
+## Running the samples
 
-The `DefaultAzureCredential` Class can be used to get a credential, and Kafka clients must use the scope of `https://<namespace>.servicebus.windows.net` to retrieve the access token from the Event Hub namespace. See [DefaultAzureCredential](https://docs.microsoft.com/en-us/python/api/azure-identity/azure.identity.defaultazurecredential?view=azure-python) for more information on this class. 
+### Configuring credentials
 
-In this example, a service principal is used to get the AAD credential by setting the following environment variables:    
+In this example, a service principal is used for the AAD credential. This service principal must have the `Azure Event Hubs Data Sender` and `Azure Event Hubs Data Receiver` roles (or equivalent) assigned on the target Event Hubs namespace. We configure `DefaultAzureCredential` by setting the following environment variables:
 
 ```shell
 export AZURE_TENANT_ID=<TenantID>
 export AZURE_CLIENT_ID=<AppClientId>
 export AZURE_CLIENT_SECRET=<AppSecret>
 ```
 
-An RBAC role must be assigned to the application to gain access to Event Hubs. Azure provides built-in roles for authorizing access to Event Hubs. See [Azure Built in Roles for Azure Event Hubs](https://docs.microsoft.com/en-us/azure/event-hubs/authorize-access-azure-active-directory#azure-built-in-roles-for-azure-event-hubs).
-
-To create a service principal and assign RBAC roles to the application, review [How to Create a Service Principal](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal).
-
-
-### Client Configuration
-
-To connect a Kafka client to Event Hub using OAuth 2.0, the following configuration is required:
+### Producing
 
-```properties
-bootstrap.servers=<namespace>.servicebus.windows.net:9093
-security.protocol=SASL_SSL
-sasl.mechanism=OAUTHBEARER
-oauth_cb=<Callback for retrieving OAuth Bearer token> 
+```shell
+# Usage: producer.py <eventhubs-namespace> <topic>.
+python producer.py mynamespace.servicebus.windows.net topic1
 ```
 
-The return value of `oauth_cb` is expected to be a (token_str, expiry_time) tuple where expiry_time is the time in seconds since the epoch as a floating point number.
-See [Confluent Kafka for Python](https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html) for reference.
-
-An example `oauth_cb` and kafka client configuration is defined below:
+> Note that the topic must already exist or else you will see an "Unknown topic or partition" error.
```

Comment:
This should be moved to the consumer section. The topic will be auto created on the producer side if it doesn't exist.

Reply:
Do you happen to know if that behavior differs across `Azure Event Hubs Data Owner` vs `Azure Event Hubs Data Sender`?

Reply:
Oh that's a good point, not sure about that. I was using data owner. Let me check with just sender

Reply:
Just tried, got the error

```
% Message failed delivery: KafkaError{code=UNKNOWN_TOPIC_OR_PART,val=3,str="Broker: Unknown topic or partition"}
```

Reply:
Ok ya same behavior on my end

Reply:
Added a note on the differing behavior between roles